src/main/scala/cognite/spark/v1/AssetHierarchyBuilder.scala

```diff
@@ -79,23 +79,40 @@ class AssetHierarchyBuilder(config: RelationConfig)(val sqlContext: SQLContext)
   import CdpConnector.ioRuntime
 
   def delete(data: DataFrame): Unit =
-    data.foreachPartition((rows: Iterator[Row]) => {
-      val deletes = rows.map(r => fromRow[DeleteItemByCogniteId](r))
-      Stream
-        .fromIterator[IO](deletes, chunkSize = batchSize)
-        .chunks
-        .parEvalMapUnordered(config.parallelismPerPartition) { chunk =>
-          client.assets
-            .deleteRecursive(
-              chunk.toVector.map(_.toCogniteId),
-              recursive = true,
-              ignoreUnknownIds = true)
-            .flatTap(_ => incMetrics(itemsDeleted, chunk.size))
-        }
-        .compile
-        .drain
-        .unsafeRunSync()
-    })
+    if (config.enableSinglePartitionDeleteAssetHierarchy) {
+      data
+        .repartition(numPartitions = 1)
+        .foreachPartition((rows: Iterator[Row]) => {
+          batchedOperation[DeleteItemByCogniteId, Unit](
+            rows.map(r => fromRow[DeleteItemByCogniteId](r)).toVector,
+            // The API calls throw exception when any of the ids do not exist
+            ids =>
+              client.assets
+                .deleteRecursive(ids.map(_.toCogniteId), recursive = true, ignoreUnknownIds = true)
+                .flatTap(_ => incMetrics(itemsDeleted, ids.size))
+                .map(Seq(_)),
+            batchSize = batchSize
+          ).void.unsafeRunSync()
```

Comment:
```suggestion
          ).void.unsafeRunSync()
```
interesting how libs author decided to use `void` name for conversion to `Unit` (parameter) type (which is not same as `Void`/`Nothing` type) :sweat_smile: